src/client.c

```diff
@@ -221,8 +221,16 @@ static bool login_via_cert(PgSocket *client)
 		goto fail;
 
 	log_debug("TLS cert login: %s", tls_peer_cert_subject(client->sbuf.tls));
-	if (!tls_peer_cert_contains_name(client->sbuf.tls, client->login_user->name)) {
-		slog_error(client, "TLS certificate name mismatch");
+
+	if (rule && rule->identmap) {
```

Comment:
Let's add similar logic to use the `map` when available to `login_as_unix_peer`, because postgres allow username mapping for that one too: https://www.postgresql.org/docs/current/auth-peer.html

(and afaict to support that we only similar 5 lines)

Reply:
I have done this change but could not add tests for it.  I was able to manually test.

I tried adding a test via: 

 result = subprocess.run([f"**sudo -u asystemuser psql -p {bouncer.port} -h  {bouncer.admin_host} p0 -c \"SELECT 1;\"**"], stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True)
    print(result.returncode, result.stdout, result.stderr)

returns error 

..... Permission denied\npsql: error: connection to server on socket **"/tmp/pytest-of-emel/pytest-34/test_client_hba_cert0/bouncer/.s.PGSQL.10202**" failed: Permission denied\n\tIs the server running locally and accepting connections on that socket?\n'

Reply:
Maybe we should make this a seperate PR with proper tests added.

Reply:
I think probably the easiest way to create a test is to manually create an ident file, that contains the current system user. So something like this (untested code):
```python
cur_user = os.current_user()
with open("myident.conf", 'w') as f:
    f.write(f"mymap {cur_user} postgres")	

bouncer.write_ini(f"auth_ident_file = pgident.conf")
```

Reply:
> Maybe we should make this a seperate PR with proper tests added.

One big reason why I'd like it in too, is because it's generally much easier to do some manual testing with it.

Reply:
Thanks @JelteF. I pushed a simple test now.